src/game/assets.rs

```diff
@@ -5,123 +5,105 @@ use bevy::{
+            ImageHandles::DUCKY.to_string(),
             asset_server.load_with_settings(
                 "images/ducky.png",
                 |settings: &mut ImageLoaderSettings| {
                     settings.sampler = ImageSampler::nearest();
                 },
             ),
         )]
-        .into()
-    }
-}
-
-#[derive(Copy, Clone, Eq, PartialEq, Hash, Reflect)]
-pub enum SfxKey {
-    ButtonHover,
-    ButtonPress,
-    Step1,
-    Step2,
-    Step3,
-    Step4,
-}
-
-impl AssetKey for SfxKey {
-    type Asset = AudioSource;
-}
-
-impl FromWorld for HandleMap<SfxKey> {
-    fn from_world(world: &mut World) -> Self {
-        let asset_server = world.resource::<AssetServer>();
-        [
-            (
-                SfxKey::ButtonHover,
-                asset_server.load("audio/sfx/button_hover.ogg"),
-            ),
-            (
-                SfxKey::ButtonPress,
-                asset_server.load("audio/sfx/button_press.ogg"),
-            ),
-            (SfxKey::Step1, asset_server.load("audio/sfx/step1.ogg")),
-            (SfxKey::Step2, asset_server.load("audio/sfx/step2.ogg")),
-            (SfxKey::Step3, asset_server.load("audio/sfx/step3.ogg")),
-            (SfxKey::Step4, asset_server.load("audio/sfx/step4.ogg")),
-        ]
-        .into()
+        .into();
+        Self(map)
     }
 }
 
-#[derive(Copy, Clone, Eq, PartialEq, Hash, Reflect)]
-pub enum SoundtrackKey {
-    Credits,
-    Gameplay,
-}
+#[derive(Resource, Debug, Deref, DerefMut, Reflect)]
+#[reflect(Resource)]
+pub struct SoundtrackHandles(HashMap<String, Handle<AudioSource>>);
 
-impl AssetKey for SoundtrackKey {
-    type Asset = AudioSource;
+impl SoundtrackHandles {
+    pub const CREDITS: &'static str = "Credits";
+    pub const GAMEPLAY: &'static str = "Gameplay";
 }
 
-impl FromWorld for HandleMap<SoundtrackKey> {
+impl FromWorld for SoundtrackHandles {
     fn from_world(world: &mut World) -> Self {
         let asset_server = world.resource::<AssetServer>();
-        [
+        let map = [
             (
-                SoundtrackKey::Credits,
+                SoundtrackHandles::CREDITS.to_string(),
                 asset_server.load("audio/soundtracks/Monkeys Spinning Monkeys.ogg"),
             ),
             (
-                SoundtrackKey::Gameplay,
+                SoundtrackHandles::GAMEPLAY.to_string(),
                 asset_server.load("audio/soundtracks/Fluffing A Duck.ogg"),
             ),
         ]
-        .into()
+        .into();
+        Self(map)
     }
 }
 
-pub trait AssetKey: Sized {
-    type Asset: Asset;
-}
-
-#[derive(Resource, Reflect, Deref, DerefMut)]
+/// This uses a map of `String` to `Vec<Handle<AudioSource>>` to store sound effects instead
+/// of the `HandleMap` used for the other assets.
+/// The key is a `String` because sound effects are often tied closely to file paths or other strings.
+/// The value is a `Vec<Handle<AudioSource>>` because a single sound effect can have multiple variations.
```

Comment:
Outdated comment. There is no `HandleMap` and everything else is using `String` as a key as well, so explaining that choice here specifically doesn't make sense. The only thing that needs to be explained is that the `Vec<_>` is for variations.

Reply:
[`212dd2e` (#225)](https://github.com/TheBevyFlock/bevy_quickstart/pull/225/commits/212dd2ea61bc0fd3c64e5022def83bdd96d5ed6c)